packages/angular_devkit/build_angular/src/tools/vite/angular-memory-plugin.ts

```diff
@@ -306,3 +314,42 @@ function pathnameWithoutBasePath(url: string, basePath: string): string {
     ? pathname.slice(basePath.length - 1)
     : pathname;
 }
+
+function angularHtmlFallbackMiddleware(
+  req: Connect.IncomingMessage,
+  res: ServerResponse,
+  next: Connect.NextFunction,
+): void {
+  // Similar to how it is handled in vite
+  // https://github.com/vitejs/vite/blob/main/packages/vite/src/node/server/middlewares/htmlFallback.ts#L15C19-L15C45
+  if (
+    // Only accept GET or HEAD
+    (req.method !== 'GET' && req.method !== 'HEAD') ||
+    // Has file extensions
+    (req.url && lookupMimeTypeFromRequest(req.url)) ||
+    // Require Accept: text/html or */*
+    !(
+      req.headers.accept === undefined || // equivalent to `Accept: */*`
+      req.headers.accept === '' || // equivalent to `Accept: */*`
+      req.headers.accept.includes('text/html') ||
+      req.headers.accept.includes('*/*')
+    )
+  ) {
+    next();
+
+    return;
+  }
+
+  req.url = '/index.html';
+  next();
```

Comment:
```suggestion
  if (
    (req.method === 'GET' || req.method === 'HEAD') &&
    (!req.url || !lookupMimeTypeFromRequest(req.url)) &&
     (!req.headers.accept ||
      req.headers.accept.includes('text/html') ||
      req.headers.accept.includes('text/*') ||
      req.headers.accept.includes('*/*')
    )
  ) {
      req.url = '/index.html';
  }

  next();
```
Consolidates to one `next()` and removes the extra `return`.
Also handles `text/*` which supports `text/html`